app/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/reportstream/ReportStreamLabOrderSenderTest.groovy

```diff
@@ -171,4 +171,71 @@ class ReportStreamLabOrderSenderTest extends Specification {
         then:
         noExceptionThrown()
     }
+
+    def "cachedPrivateKey getter and setter works" () {
+        given:
+        def rsLabOrderSender = ReportStreamLabOrderSender.getInstance()
+        def expected = "a fake azure key"
+
+        when:
+        rsLabOrderSender.setCachedPrivateKey(expected)
+        def actual = rsLabOrderSender.getCachedPrivateKey()
+
+        then:
+        expected == actual
+    }
+
+    def "cachedPrivateKey thread synchronization is working" () {
+        given:
+        def rsLabOrderSender = ReportStreamLabOrderSender.getInstance()
+        def threadCount = 15
+        def lock = new Object()
+        def threads = (1..threadCount).collect { index ->
+            new Thread( {
+                synchronized (lock) {
```

Comment:
What is this scenario really intended to test? Won't the key normally be the same value per execution environment? (which may mitigate the impact of race conditions)

Also, shouldn't synchronization be done within rsLabOrderSender to represent a realistic use case?